With the fix, the linker failed in the FastTest CI job:

```
2022-10-03 00:23:09 [4133/4148] Linking CXX executable programs/clickhouse
2022-10-03 00:23:09 FAILED: programs/clickhouse 
2022-10-03 00:23:09 : && /usr/bin/clang++-15 [...] /clickhouse.debug
2022-10-03 00:23:09 ld.lld-15: error: unable to find library -lclickhouse_core
2022-10-03 00:23:09 clang: error: linker command failed with exit code 1 (use -v to see invocation)
2022-10-03 00:23:09 ninja: build stopped: subcommand failed.
```

FastTest does a standard static library build. I think the problem is that "clickhouse_core" exists as a separate entity only in shared library builds, see macro `add_object_library` in "src/CMakeLists.txt". We could link "dbms" instead, which exists in both build types. But since we need it only for Sentry, we also shouldn't link it in the general case. Here's a proposal:

```
if (TARGET ch_contrib::sentry)
    target_link_libraries (daemon PRIVATE ch_contrib::sentry dbms)
endif ()
```

--- a/src/Daemon/CMakeLists.txt
+++ b/src/Daemon/CMakeLists.txt
@@ -14,5 +14,5 @@ endif()
 target_link_libraries (daemon PUBLIC loggers common PRIVATE clickhouse_common_io clickhouse_common_config ${EXECINFO_LIBRARIES})
 
 if (TARGET ch_contrib::sentry)
-    target_link_libraries (daemon PRIVATE ch_contrib::sentry)
+    target_link_libraries (daemon PRIVATE ch_contrib::sentry dbms)
 endif ()

